src/core/lib/transport/metadata.cc

```diff
@@ -211,7 +211,7 @@ void grpc_mdctx_global_shutdown() {
     mdtab_shard* shard = &g_shards[i];
     gpr_mu_destroy(&shard->mu);
     gc_mdtab(shard);
-    /* TODO(ctiller): GPR_ASSERT(shard->count == 0); */
+    GPR_DEBUG_ASSERT(shard->count == 0);
```

Comment:
Hmm, actually, it looks like we already have a way to trigger this failure, using the `grpc_iomgr_abort_on_leaks()` mechanism on line 218 below.  Maybe we should just make sure that we're enabling that in tests?

Reply:
GRPC_ABORT_ON_LEAKS=1 in the environment is what we need. Where are the test env vars specified?

Reply:
Hmm... Actually, it looks like `grpc_iomgr_abort_on_leaks()` controls more than just this, and there are probably other cases where it would cause failures if we enabled it for every test.

Instead, I suggest just moving this assertion down to line 222.  That way, we'll get the warning message before this assertion triggers, and we'll still get the `grpc_iomgr_abort_on_leaks()` behavior even in non-debug builds.

Reply:
Done.

Reply:
I don't see the change.

Reply:
My mistake, forgot to push.